packages/ui-kit/src/components/lord-icon-client-side.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useRef, useState } from 'react';
+import Lottie, { LottieRefCurrentProps } from 'lottie-react';
+
+type LordIconProps = {
+  iconName: string;
+  loop: boolean;
+  autoplay: boolean;
+  style: object;
+};
+
+const defaultProps = {
+  loop: false,
+  autoplay: false,
+};
+
+const LordIcon = (props: LordIconProps) => {
+  const [animationData, setAnimationData] = useState();
+  const lottieRef = useRef<LottieRefCurrentProps>(null);
+  const { iconName, ...rest } = props;
+
+  useEffect(() => {
+    const loadAnimationData = async () => {
+      try {
+        const data = await import(
```

Comment:
Did you consider importing and holding the fixed set of icons that we know are used anyways statically in the module?    At least for the top nav content we do know that every single website will need and load it anyways.   Could keep the dynamic import for other icons that are in the distribution.     Basically like you do it for the preview SVG, they are statically in the server side build and fall back to nothing. 

(Maybe that would resolve the flicker, too? )